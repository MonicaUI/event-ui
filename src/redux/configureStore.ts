import analyticsConfig from './analytics';
import reducer from './reducer';

// using require instead of import for conditional compilation;
let createStoreWithMiddleware;
if (process.env.NODE_ENV === 'production') {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  createStoreWithMiddleware = require('nucleus-guestside-site/src/redux/prodCreateStoreWithMiddleware').default;
} else {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  createStoreWithMiddleware = require('nucleus-guestside-site/src/redux/devCreateStoreWithMiddleware').default;
}

export default function configureStore(
  initialState: $TSFixMe,
  baseAnalyticsConfig = {},
  thunkExtraArgument?: $TSFixMe
): $TSFixMe {
  return createStoreWithMiddleware(reducer, initialState, {
    analyticsConfig: analyticsConfig(initialState, baseAnalyticsConfig),
    thunkExtraArgument
  });
}
