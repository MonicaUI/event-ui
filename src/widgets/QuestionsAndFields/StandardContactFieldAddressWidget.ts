import { connect } from 'react-redux';
import StandardContactFieldAddressWidget from 'event-widgets/lib/StandardContactFields/StandardContactFieldAddressWidget';
import StandardContactFields from 'event-widgets/lib/StandardContactFields/StandardContactFields';
import { loadCountryStates } from '../../redux/states';
import { getLocalizedContactFieldForWidget } from './utils';
import { getCountries, getCountryCodesJsonPath } from '../../redux/selectors/shared';
import { getEventRegistrationId } from '../../redux/selectors/currentRegistrant';
import * as eventRegistrationData from '../../utils/eventRegistrationData';
import { defaultMemoize } from 'reselect';
import withMappedWidgetConfig from 'nucleus-widgets/lib/withMappedWidgetConfig';
import { getIn } from 'icepick';
import { evaluateQuestionVisibilityLogic } from '../../redux/actions';
import { COUNTRY_LIST_RELEASE_VARIANT } from '@cvent/event-ui-experiments';
import getJSONValue from 'nucleus-widgets/utils/fields/getJSONValue';

const EMPTY_OBJECT = Object.freeze({});

export function getFilteredCountries(state: $TSFixMe, widgetId: $TSFixMe, config: $TSFixMe): $TSFixMe {
  const filteredCountries = {};
  const countryListPath = getCountryCodesJsonPath(
    state,
    widgetId,
    config.registrationFieldPageType,
    config.fieldId,
    'countryCodes'
  );
  const countryList = getJSONValue(state.appData, countryListPath);
  const defaultCountries = getCountries(state);

  if (countryList?.length > 0) {
    Object.keys(defaultCountries)
      .filter(code => code === '' || countryList.includes(code))
      .forEach(code => (filteredCountries[code] = defaultCountries[code]));
    return filteredCountries;
  }
  return defaultCountries;
}

const setDefaultCountry = (countryCode, defaultCountryCode, onAddressChange) => {
  return () => {
    if (!countryCode && defaultCountryCode) {
      if (onAddressChange) {
        onAddressChange('countryCode', defaultCountryCode);
      }
    }
  };
};

function getRegistrationField(state, config, id) {
  const registrationField = getLocalizedContactFieldForWidget(state, config, id);
  return {
    ...registrationField,
    subFieldDisplayNames: {
      address1: getIn(config, ['appData', 'address1']) || registrationField.subFieldDisplayNames.address1,
      address2: getIn(config, ['appData', 'address2']) || registrationField.subFieldDisplayNames.address2,
      address3: getIn(config, ['appData', 'address3']) || registrationField.subFieldDisplayNames.address3,
      city: getIn(config, ['appData', 'city']) || registrationField.subFieldDisplayNames.city,
      country: getIn(config, ['appData', 'country']) || registrationField.subFieldDisplayNames.country,
      state: getIn(config, ['appData', 'state']) || registrationField.subFieldDisplayNames.state,
      zip: getIn(config, ['appData', 'zip']) || registrationField.subFieldDisplayNames.zip
    }
  };
}

/**
 * Data wrapper for the event standard contact field address widget.
 */
export default withMappedWidgetConfig(
  connect(
    () => {
      const buildEventRegistrationPath = defaultMemoize(fieldPath => {
        return eventRegistrationData.buildEventRegistrationPath(fieldPath);
      });

      const getAnswer = eventRegistrationData.createAnswer();
      const memoized = {
        getRegistrationField: defaultMemoize(getRegistrationField)
      };

      return (state: $TSFixMe, props: $TSFixMe) => {
        const registrationField = memoized.getRegistrationField(state, props.config, props.id);

        const fieldPath = StandardContactFields[props.config.fieldId].regApiPath;

        const answer = getAnswer({
          state,
          widgetConfig: props.config,
          eventRegistrationPath: buildEventRegistrationPath(fieldPath),
          getAnswerFormatter: v => v || EMPTY_OBJECT
        });

        // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
        const isWidgetPlacedOnGuestModal = (answer && answer.isWidgetPlacedOnGuestModal) || false;

        return {
          values: answer.value,
          setterAction: answer.setterAction,
          registrationField,
          countries:
            state.experiments?.featureRelease >= COUNTRY_LIST_RELEASE_VARIANT
              ? getFilteredCountries(state, props.id, props.config)
              : getCountries(state),
          selectedCountryStates: state.states,
          defaultCountryCode: registrationField.defaultCountry.value,
          eventRegistrationId: getEventRegistrationId(state),
          skipRequiredValidation: state.defaultUserSession.isPlanner,
          isWidgetPlacedOnGuestModal
        };
      };
    },
    {
      loadCountryStates,
      setAnswerAction: eventRegistrationData.setAnswerAction,
      setDefaultCountry,
      evaluateQuestionVisibilityLogic
    },
    (stateProps: $TSFixMe, dispatchProps: $TSFixMe, ownProps: $TSFixMe) => {
      return {
        ...ownProps,
        ...stateProps,
        ...dispatchProps,
        onAddressChange: (fieldName, value) => {
          const modifiedField = fieldName.split('.')[0] || fieldName;
          let updatedValue = value;
          if (fieldName.startsWith('countryCode')) {
            updatedValue = value || stateProps.defaultCountryCode;
            const updatedCountry = updatedValue ? stateProps.countries[updatedValue].name : '';
            dispatchProps.setAnswerAction(stateProps.setterAction, updatedCountry, 'country');
          } else if (fieldName.startsWith('stateCode')) {
            const selectedCountry = stateProps?.values?.countryCode;
            const states = stateProps?.selectedCountryStates?.[selectedCountry]?.states;
            if (selectedCountry && states) {
              const selectedState = states?.[updatedValue]?.name;
              const updatedState = selectedState || '';
              dispatchProps.setAnswerAction(stateProps.setterAction, updatedState, 'state');
            }
          }
          if (
            stateProps.defaultCountryCode &&
            !stateProps.values?.countryCode &&
            !stateProps.values?.country &&
            !fieldName.startsWith('country')
          ) {
            // if for whatever reason (e.g. after "clear fields") the country in the store is blank,
            // also reset it to default country
            dispatchProps.setAnswerAction(stateProps.setterAction, stateProps.defaultCountryCode, 'countryCode');
            dispatchProps.setAnswerAction(
              stateProps.setterAction,
              stateProps.countries[stateProps.defaultCountryCode]?.name,
              'country'
            );
            loadCountryStates(stateProps.defaultCountryCode);
          }
          dispatchProps.setAnswerAction(stateProps.setterAction, updatedValue, modifiedField);
        }
      };
    }
    // @ts-expect-error ts-migrate(2345) FIXME: Argument of type 'typeof StandardContactFieldAddre... Remove this comment to see the full error message
  )(StandardContactFieldAddressWidget)
);
