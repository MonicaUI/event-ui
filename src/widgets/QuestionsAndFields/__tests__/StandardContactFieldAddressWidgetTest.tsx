import React from 'react';
import StandardContactFieldAddressWidget from '../StandardContactFieldAddressWidget';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from '../../../redux/configureStore';
import { getRegCart } from '../../../redux/selectors/shared';
import { CLEAR_REG_CART_INFERRED_FIELDS } from '../../../redux/registrationForm/regCart/actionTypes';
import WidgetFactory from '../../../widgetFactory';
import Grid from 'nucleus-core/layout/flexbox/Grid';
import * as registrationFieldPageType from 'event-widgets/utils/registrationFieldPageType';
import pageContainingWidgetFixture from '../../../testUtils/pageContainingWidgetFixture';
import { setIn } from 'icepick';
import { getFilteredCountries } from '../StandardContactFieldAddressWidget';

const HOME_ADDRESS_FIELD_ID = 'b5a8362f-91e7-46ff-a704-20bfb3ca6ce5';

const getInitialState = (defaultCountry?, countryList?) => ({
  website: {
    ...pageContainingWidgetFixture('pageId', 'widgetId'),
    pluginData: {
      registrationProcessNavigation: {
        registrationPaths: {
          regPathId: {
            id: 'regPathId',
            pageIds: ['pageId']
          }
        }
      }
    },
    siteInfo: {
      sharedConfigs: {}
    }
  },
  widgetFactory: new WidgetFactory(),
  appData: {
    registrationSettings: {
      registrationPaths: {
        regPathId: {
          registrationPageFields: {
            [registrationFieldPageType.Registration]: {
              registrationFieldPageType: registrationFieldPageType.Registration,
              homeAddressSettings: {
                countryCodes: countryList
              },
              registrationFields: {
                [HOME_ADDRESS_FIELD_ID]: {
                  fieldName: 'Home Address',
                  fieldId: HOME_ADDRESS_FIELD_ID,
                  displayName: 'Home Address',
                  display: 2,
                  isCustomField: false,
                  defaultCountry: {
                    value: defaultCountry
                  },
                  subFieldDisplayNames: {
                    address1: 'Address 1',
                    address2: 'Address 2',
                    address3: 'Address 3',
                    city: 'City',
                    country: 'Country',
                    state: 'State',
                    zip: 'Zip'
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  registrationForm: {
    currentEventRegistrationId: 'eventRegistration1',
    regCart: {
      eventRegistrations: {
        eventRegistration1: {
          registrationPathId: 'regPathId'
        }
      }
    }
  },
  countries: {
    countries: {
      '': { nameResourceKey: '', name: '', code: '' },
      US: { nameResourceKey: 'us_name_resx', name: 'US', code: 'US' },
      GB3: { nameResourceKey: 'wales_name_resx', name: 'Wales', code: 'GB3' }
    }
  },
  states: {
    US: {
      states: {
        '': {},
        AL: { name: 'Alabama' }
      }
    }
  }
});

const initialState = getInitialState('US', []);

const defaultProps = {
  translate: (resx, opts) => (opts ? `${resx}:${JSON.stringify(opts)}` : resx),
  style: {},
  classes: {},
  id: 'widgetId',
  disabled: false
};

test('address standard contact field renders and allows you to enter values', async () => {
  const store = configureStore(initialState);
  const component = mount(
    <Provider store={store}>
      <Grid>
        <StandardContactFieldAddressWidget
          {...defaultProps}
          config={{
            fieldId: HOME_ADDRESS_FIELD_ID,
            registrationFieldPageType: registrationFieldPageType.Registration
          }}
          type="EventStandardContactFieldText"
        />
      </Grid>
    </Provider>
  );

  // unanswered view
  expect(component).toMatchSnapshot();
  expect(store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee).toMatchSnapshot();

  // enter text
  component.find('[name^="address1"]').simulate('change', { target: { value: 'address 1 answer' } });
  component.find('[name^="countryCode"]').simulate('change', { target: { value: '2' } });
  expect(component).toMatchSnapshot();
  expect(store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee).toMatchSnapshot();
});

test('address standard contact field renders default country name on selecting empty country when default country is set', async () => {
  const store = configureStore(initialState);
  const component = mount(
    <Provider store={store}>
      <Grid>
        <StandardContactFieldAddressWidget
          {...defaultProps}
          config={{
            fieldId: HOME_ADDRESS_FIELD_ID,
            registrationFieldPageType: registrationFieldPageType.Registration
          }}
          type="EventStandardContactFieldText"
        />
      </Grid>
    </Provider>
  );
  component.find('[name^="countryCode"]').simulate('change', { target: { value: '0' } });
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.country
  ).toBe('US');
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.countryCode
  ).toBe('US');
});

test('address standard contact field renders no country name on selecting empty country when default country is not set', async () => {
  const nonDefaultCountryCodeState = setIn(
    initialState,
    [
      'appData',
      'registrationSettings',
      'registrationPaths',
      'regPathId',
      'registrationPageFields',
      registrationFieldPageType.Registration,
      'registrationFields',
      HOME_ADDRESS_FIELD_ID,
      'defaultCountry',
      'value'
    ],
    ''
  );

  const store = configureStore(nonDefaultCountryCodeState);
  const component = mount(
    <Provider store={store}>
      <Grid>
        <StandardContactFieldAddressWidget
          {...defaultProps}
          config={{
            fieldId: HOME_ADDRESS_FIELD_ID,
            registrationFieldPageType: registrationFieldPageType.Registration
          }}
          type="EventStandardContactFieldText"
        />
      </Grid>
    </Provider>
  );
  component.find('[name^="countryCode"]').simulate('change', { target: { value: '0' } });
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.country
  ).toBe('');
});

test('address standard contact field records selected state name on selecting a state', async () => {
  const store = configureStore(initialState);
  const component = mount(
    <Provider store={store}>
      <Grid>
        <StandardContactFieldAddressWidget
          {...defaultProps}
          config={{
            fieldId: HOME_ADDRESS_FIELD_ID,
            registrationFieldPageType: registrationFieldPageType.Registration
          }}
          type="EventStandardContactFieldText"
        />
      </Grid>
    </Provider>
  );
  component.find('[name^="stateCode"]').simulate('change', { target: { value: '1' } });
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.state
  ).toBe('Alabama');
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.stateCode
  ).toBe('AL');

  component.find('[name^="stateCode"]').simulate('change', { target: { value: '0' } });
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.state
  ).toBe('');
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.stateCode
  ).toBe('');
});

test('address standard contact field resets blank country to default on address change', async () => {
  const store = configureStore(initialState);
  const component = mount(
    <Provider store={store}>
      <Grid>
        <StandardContactFieldAddressWidget
          {...defaultProps}
          config={{
            fieldId: HOME_ADDRESS_FIELD_ID,
            registrationFieldPageType: registrationFieldPageType.Registration
          }}
          type="EventStandardContactFieldText"
        />
      </Grid>
    </Provider>
  );

  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.countryCode
  ).toBe('US');
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.country
  ).toBe('US');

  // clear fields
  store.dispatch({
    type: CLEAR_REG_CART_INFERRED_FIELDS,
    payload: {
      regCart: setIn(
        getRegCart(store.getState()),
        ['eventRegistrations', 'eventRegistration1', 'attendee', 'personalInformation'],
        {}
      )
    }
  });

  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress
  ).toBeUndefined();

  // simulate address change
  component.find('[name^="address1"]').simulate('change', { target: { value: 'address 1 answer' } });

  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.countryCode
  ).toBe('US');
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.country
  ).toBe('US');
});

test("address standard contact field doesn't reset country when no default value set", async () => {
  const state = getInitialState();
  const store = configureStore(state);
  const component = mount(
    <Provider store={store}>
      <Grid>
        <StandardContactFieldAddressWidget
          {...defaultProps}
          config={{
            fieldId: HOME_ADDRESS_FIELD_ID,
            registrationFieldPageType: registrationFieldPageType.Registration
          }}
          type="EventStandardContactFieldText"
        />
      </Grid>
    </Provider>
  );

  // clear fields
  store.dispatch({
    type: CLEAR_REG_CART_INFERRED_FIELDS,
    payload: {
      regCart: setIn(
        getRegCart(store.getState()),
        ['eventRegistrations', 'eventRegistration1', 'attendee', 'personalInformation'],
        {}
      )
    }
  });

  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress
  ).toBeUndefined();

  // simulate address change
  component.find('[name^="address1"]').simulate('change', { target: { value: 'address 1 answer' } });

  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.countryCode
  ).toBeUndefined();
  expect(
    store.getState().registrationForm.regCart.eventRegistrations.eventRegistration1.attendee.personalInformation
      .homeAddress.country
  ).toBeUndefined();
});

test('getFilteredCountries function must return countries in homeAddressSettings only', async () => {
  const state = getInitialState('US', ['US']);
  const props = {
    ...defaultProps,
    config: {
      fieldId: HOME_ADDRESS_FIELD_ID,
      registrationFieldPageType: registrationFieldPageType.Registration
    }
  };
  const result = getFilteredCountries(state, props.id, props.config);
  const expectedResult = {
    '': { nameResourceKey: '', name: '', code: '' },
    US: { nameResourceKey: 'us_name_resx', name: 'US', code: 'US' }
  };
  expect(result).toStrictEqual(expectedResult);
});

test('getFilteredCountries function should return default contries if country list is empty', async () => {
  const state = getInitialState();
  const props = {
    ...defaultProps,
    config: {
      fieldId: HOME_ADDRESS_FIELD_ID,
      registrationFieldPageType: registrationFieldPageType.Registration
    }
  };
  const result = getFilteredCountries(state, props.id, props.config);
  const expectedResult = {
    '': { nameResourceKey: '', name: '', code: '' },
    US: { nameResourceKey: 'us_name_resx', name: 'US', code: 'US' },
    GB3: { nameResourceKey: 'wales_name_resx', name: 'Wales', code: 'GB3' }
  };
  expect(result).toStrictEqual(expectedResult);
});
