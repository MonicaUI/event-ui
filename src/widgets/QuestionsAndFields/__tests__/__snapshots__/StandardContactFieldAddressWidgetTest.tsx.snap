// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address standard contact field renders and allows you to enter values 1`] = `
<div
  class=""
>
  <div
    class=""
    style="width: 100%;"
  >
    <div
      class=""
    >
      <div
        class=""
        style="flex: 1 1 auto;"
      >
        <legend
          class=""
        >
          Home Address
        </legend>
        <div
          class=""
        >
          <div
            class=""
          >
            <label
              class="labelOnTop"
              for="address1.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
            >
              <span
                class=""
              >
                <span
                  class=""
                >
                  * 
                </span>
                <span
                  data-cvent-id="label"
                >
                  <span>
                    Address 1
                  </span>
                </span>
              </span>
            </label>
            <div
              class=""
            >
              <div
                class=""
              >
                <input
                  class=""
                  data-cvent-id="input"
                  id="address1.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  maxlength="40"
                  name="address1.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  placeholder=""
                  required=""
                  style="opacity: 1;"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <label
              class="labelOnTop"
              for="address2.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
            >
              <span
                class=""
              >
                <span
                  data-cvent-id="label"
                >
                  <span>
                    Address 2
                  </span>
                </span>
              </span>
            </label>
            <div
              class=""
            >
              <div
                class=""
              >
                <input
                  class=""
                  data-cvent-id="input"
                  id="address2.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  maxlength="40"
                  name="address2.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  placeholder=""
                  style="opacity: 1;"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <label
              class="labelOnTop"
              for="address3.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
            >
              <span
                class=""
              >
                <span
                  data-cvent-id="label"
                >
                  <span>
                    Address 3
                  </span>
                </span>
              </span>
            </label>
            <div
              class=""
            >
              <div
                class=""
              >
                <input
                  class=""
                  data-cvent-id="input"
                  id="address3.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  maxlength="40"
                  name="address3.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  placeholder=""
                  style="opacity: 1;"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <label
                class="labelOnTop"
                for="countryCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
              >
                <span
                  class=""
                >
                  <span
                    class=""
                  >
                    * 
                  </span>
                  <span
                    data-cvent-id="label"
                  >
                    <span>
                      Country
                    </span>
                  </span>
                </span>
              </label>
              <div
                class=""
              >
                <select
                  class=""
                  data-cvent-id="input"
                  id="countryCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  name="countryCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  required=""
                  style="opacity: 1; width: 100%;"
                >
                  <option
                    data-cvent-id="option-0"
                    value="0"
                  />
                  <option
                    data-cvent-id="option-1"
                    value="1"
                  >
                    us_name_resx
                  </option>
                  <option
                    data-cvent-id="option-2"
                    value="2"
                  >
                    wales_name_resx
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <div
                class=""
              >
                <label
                  class="labelOnTop"
                  for="city.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                >
                  <span
                    class=""
                  >
                    <span
                      class=""
                    >
                      * 
                    </span>
                    <span
                      data-cvent-id="label"
                    >
                      <span>
                        City
                      </span>
                    </span>
                  </span>
                </label>
                <div
                  class=""
                >
                  <div
                    class=""
                  >
                    <input
                      class=""
                      data-cvent-id="input"
                      id="city.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                      maxlength="40"
                      name="city.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                      placeholder=""
                      required=""
                      style="opacity: 1;"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class=""
            >
              <div
                class=""
              >
                <label
                  class="labelOnTop"
                  for="stateCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                >
                  <span
                    class=""
                  >
                    <span
                      class=""
                    >
                      * 
                    </span>
                    <span
                      data-cvent-id="label"
                    >
                      <span>
                        State
                      </span>
                    </span>
                  </span>
                </label>
                <div
                  class=""
                >
                  <select
                    class=""
                    data-cvent-id="input"
                    id="stateCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    name="stateCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    required=""
                    style="opacity: 1; width: 100%;"
                  >
                    <option
                      data-cvent-id="option-0"
                      value="0"
                    />
                    <option
                      data-cvent-id="option-1"
                      value="1"
                    />
                  </select>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <label
                class="labelOnTop"
                for="postalCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
              >
                <span
                  class=""
                >
                  <span
                    class=""
                  >
                    * 
                  </span>
                  <span
                    data-cvent-id="label"
                  >
                    <span>
                      Zip
                    </span>
                  </span>
                </span>
              </label>
              <div
                class=""
              >
                <div
                  class=""
                >
                  <input
                    class=""
                    data-cvent-id="input"
                    id="postalCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    maxlength="25"
                    name="postalCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    placeholder=""
                    required=""
                    style="opacity: 1;"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`address standard contact field renders and allows you to enter values 2`] = `
Object {
  "personalInformation": Object {
    "homeAddress": Object {
      "country": "US",
      "countryCode": "US",
    },
  },
}
`;

exports[`address standard contact field renders and allows you to enter values 3`] = `
<div
  class=""
>
  <div
    class=""
    style="width: 100%;"
  >
    <div
      class=""
    >
      <div
        class=""
        style="flex: 1 1 auto;"
      >
        <legend
          class=""
        >
          Home Address
        </legend>
        <div
          class=""
        >
          <div
            class=""
          >
            <label
              class="labelOnTop"
              for="address1.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
            >
              <span
                class=""
              >
                <span
                  class=""
                >
                  * 
                </span>
                <span
                  data-cvent-id="label"
                >
                  <span>
                    Address 1
                  </span>
                </span>
              </span>
            </label>
            <div
              class=""
            >
              <div
                class=""
              >
                <input
                  class=""
                  data-cvent-id="input"
                  id="address1.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  maxlength="40"
                  name="address1.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  placeholder=""
                  required=""
                  style="opacity: 1;"
                  type="text"
                  value="address 1 answer"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <label
              class="labelOnTop"
              for="address2.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
            >
              <span
                class=""
              >
                <span
                  data-cvent-id="label"
                >
                  <span>
                    Address 2
                  </span>
                </span>
              </span>
            </label>
            <div
              class=""
            >
              <div
                class=""
              >
                <input
                  class=""
                  data-cvent-id="input"
                  id="address2.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  maxlength="40"
                  name="address2.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  placeholder=""
                  style="opacity: 1;"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <label
              class="labelOnTop"
              for="address3.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
            >
              <span
                class=""
              >
                <span
                  data-cvent-id="label"
                >
                  <span>
                    Address 3
                  </span>
                </span>
              </span>
            </label>
            <div
              class=""
            >
              <div
                class=""
              >
                <input
                  class=""
                  data-cvent-id="input"
                  id="address3.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  maxlength="40"
                  name="address3.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  placeholder=""
                  style="opacity: 1;"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <label
                class="labelOnTop"
                for="countryCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
              >
                <span
                  class=""
                >
                  <span
                    class=""
                  >
                    * 
                  </span>
                  <span
                    data-cvent-id="label"
                  >
                    <span>
                      Country
                    </span>
                  </span>
                </span>
              </label>
              <div
                class=""
              >
                <select
                  class=""
                  data-cvent-id="input"
                  id="countryCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  name="countryCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                  required=""
                  style="opacity: 1; width: 100%;"
                >
                  <option
                    data-cvent-id="option-0"
                    value="0"
                  />
                  <option
                    data-cvent-id="option-1"
                    value="1"
                  >
                    us_name_resx
                  </option>
                  <option
                    data-cvent-id="option-2"
                    value="2"
                  >
                    wales_name_resx
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <div
                class=""
              >
                <label
                  class="labelOnTop"
                  for="city.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                >
                  <span
                    class=""
                  >
                    <span
                      class=""
                    >
                      * 
                    </span>
                    <span
                      data-cvent-id="label"
                    >
                      <span>
                        City
                      </span>
                    </span>
                  </span>
                </label>
                <div
                  class=""
                >
                  <div
                    class=""
                  >
                    <input
                      class=""
                      data-cvent-id="input"
                      id="city.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                      maxlength="40"
                      name="city.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                      placeholder=""
                      required=""
                      style="opacity: 1;"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class=""
            >
              <div
                class=""
              >
                <label
                  class="labelOnTop"
                  for="stateCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                >
                  <span
                    class=""
                  >
                    <span
                      class=""
                    >
                      * 
                    </span>
                    <span
                      data-cvent-id="label"
                    >
                      <span>
                        State
                      </span>
                    </span>
                  </span>
                </label>
                <div
                  class=""
                >
                  <select
                    class=""
                    data-cvent-id="input"
                    id="stateCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    name="stateCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    required=""
                    style="opacity: 1; width: 100%;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <div
              class=""
            >
              <label
                class="labelOnTop"
                for="postalCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
              >
                <span
                  class=""
                >
                  <span
                    class=""
                  >
                    * 
                  </span>
                  <span
                    data-cvent-id="label"
                  >
                    <span>
                      Zip
                    </span>
                  </span>
                </span>
              </label>
              <div
                class=""
              >
                <div
                  class=""
                >
                  <input
                    class=""
                    data-cvent-id="input"
                    id="postalCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    maxlength="25"
                    name="postalCode.b5a8362f-91e7-46ff-a704-20bfb3ca6ce5"
                    placeholder=""
                    required=""
                    style="opacity: 1;"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`address standard contact field renders and allows you to enter values 4`] = `
Object {
  "personalInformation": Object {
    "homeAddress": Object {
      "address1": "address 1 answer",
      "country": "Wales",
      "countryCode": "GB3",
      "state": "",
      "stateCode": "",
    },
  },
}
`;
